### PR TITLE
Automated creation of JIRA issues based on items

### PR DESCRIPTION
**Pull request by JasperCraeghs**
- [x] Added documentation
- [x] Added unit tests

### CODE REVIEW

File: doc/usage.rst [hunk 1]
```diff
@@ -480,3 +480,56 @@ GitHub
...
 .. _`personal access token`: https://github.blog/2013-05-16-personal-api-tokens/
+
+.. _traceability_jira_automation:
+
+--------------------
+Jira ticket creation
+--------------------
+
+Traceable items can be configured to create Jira tickets for. Duplication of tickets is avoided by querying Jira
+first for existing tickets based on the Jira project and ticket summary. Below is an example configuration:
```
Comment (Letme) [inline doc/usage.rst:491]: Can we make field choice (`summary`) as configurable in below configuration?

File: tests/test_jira_interaction.py [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+            ['WARNING:root:Dummy log']
+        )
+
+        self.assertEqual(jira.call_args,
+                         mock.call({'server': 'https://jira.atlassian.com/rest/api/latest/'},
+                                   basic_auth=('my_username', 'my_password')))
+        self.assertEqual(jira_mock.search_issues.call_args_list,
+                         [
+                             mock.call("project=MLX12345 and summary ~ 'MEETING-12345_2 Caption for action 1'"),
+                             mock.call("project=MLX12345 and summary ~ 'Caption for action 2'"),
```
Comment (Letme) [inline tests/test_jira_interaction.py:133]: Interesting solution, I was mostly looking at `MagicMock` and `@patch` for such things.

Comment (JasperCraeghs) [reply tests/test_jira_interaction.py:133]: `jira_mock` is a `MagicMock` instance and `@patch` is used on the entire test suite. How else could I have done it?

Comment (Letme) [reply tests/test_jira_interaction.py:133]: Hm, ok I missed that. Didn't know you can apply `@patch` to whole suite anyway. Cool.

File: doc/usage.rst [hunk 3]
```diff
@@ -480,3 +480,56 @@ GitHub
...
+
+    traceability_jira_automation = {
+        'api_endpoint': 'https://jira.atlassian.com/rest/api/latest/',
+        'username': 'my_username',
+        'password': 'my_password',
+        'project_key_regexp': r'ACTION-(?P<project>\d{5})_',
+        'project_key_prefix': 'MLX',
+        'default_project': 'SWCC',
+        'issue_type': 'Task',
+        'item_to_issue_regex': r'ACTION-12345_ACTION_\d+',
```
Comment (JasperCraeghs) [inline doc/usage.rst:506]: Would a key `'item_to_ticket_regex'` be a better fit?

Comment (Letme) [reply doc/usage.rst:506]: Yes, since jira jargon is ticket

Comment (JasperCraeghs) [reply doc/usage.rst]: Great, done in cc3aa7da57e3a40c146fd731df0dba4ffb82dbcf

File: doc/usage.rst (hunk 1 above)
Comment (JasperCraeghs) [reply doc/usage.rst]: Updated docs in 9dc42e940c7c5caf58b77a382207a7ccb800e356 and implementation in f739f7646fb80e22c1128bda067ace787b5ee470

File: doc/usage.rst [hunk 4]
```diff
@@ -480,3 +480,56 @@ GitHub
...
+        'warn_if_existent': True,
```
[the remark this answers is not in the document]
Comment (JasperCraeghs) [reply doc/usage.rst]: Done in 5c35a68d7cc4c9c5bdbfcb5145cd66bd5c2ac356

File: doc/usage.rst [hunk 5]
```diff
@@ -480,3 +480,56 @@ GitHub
...
 
 .. _`personal access token`: https://github.blog/2013-05-16-personal-api-tokens/
+
+.. _traceability_jira_automation:
+
+--------------------
+Jira ticket creation
+--------------------
+
+Traceable items can be configured to create Jira tickets for. Duplication of tickets is avoided by querying Jira
```
[the remark this answers is not in the document]
Comment (JasperCraeghs) [reply doc/usage.rst]: Your suggestion is not what I was aiming for. I tweaked this paragraph in 9dc42e940c7c5caf58b77a382207a7ccb800e356

**Pull request merged**